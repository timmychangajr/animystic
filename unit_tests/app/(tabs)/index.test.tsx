import HomeScreen from '@/app/(tabs)/index';
import { cleanup, fireEvent, render, renderHook } from '@testing-library/react-native';
import { useRouter } from 'expo-router';

describe('HomeScreen', () => {
  afterEach(cleanup);
  const initSearch = async () => {
    const navigate = renderHook(() => useRouter())
    navigate.result.current.navigate = jest.fn()
    const tree = render(<HomeScreen />);
    const input = tree.getByTestId('search_input');
    const button = tree.getByTestId('search_btn');
    return { tree, input, button }
  }
  test('renders correctly', () => {
    const tree = render(<HomeScreen />);
    expect(tree).toMatchSnapshot();
  });
  test('should display recent search on submission', async () => {
    const { tree, input, button } = await initSearch()
    
    fireEvent.changeText(input, 'sloth');
    fireEvent.press(button);
    
    await tree.findByTestId('recent_view')
    const recentButton = tree.getByTestId('recent_btn');
    fireEvent.press(recentButton);
  });
  test('should display error in placeholder with invalid search', async () => {
    const { tree, input, button } = await initSearch()

    fireEvent.changeText(input, 'pickle');
    fireEvent.press(button);

    await tree.findByPlaceholderText('Try another animal...')
  });
    test('should search animal unavailable data', async () => {
    const { tree, input, button } = await initSearch()

    fireEvent.changeText(input, 'monkey');
    fireEvent.press(button);

    await tree.findByPlaceholderText('Try another animal...')
  });
});