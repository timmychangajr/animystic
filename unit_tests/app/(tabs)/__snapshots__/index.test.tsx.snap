// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      false,
      {
        "backgroundColor": "#000000B5",
      },
      {
        "backgroundColor": "#000000B5",
        "flex": 1,
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 20,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "padding": 20,
          "paddingVertical": 10,
        },
        {
          "flexDirection": "column",
          "paddingVertical": 20,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontFamily": "Outfit",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "fontSize": 25,
              "lineHeight": 32,
            },
            undefined,
            undefined,
          ]
        }
      >
        Animystic
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 10,
          "paddingTop": 20,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/ani_sloth.png",
          }
        }
        style={
          {
            "borderRadius": 20,
            "height": 50,
            "width": 50,
          }
        }
      />
      <Image
        source={
          {
            "testUri": "../../../assets/images/ani_monkey.png",
          }
        }
        style={
          {
            "borderRadius": 20,
            "height": 50,
            "width": 50,
          }
        }
      />
      <Image
        source={
          {
            "testUri": "../../../assets/images/ani_frog.png",
          }
        }
        style={
          {
            "borderRadius": 20,
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
    style={
      {
        "flex": 1,
        "overflow": "scroll",
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          {
            "paddingBottom": 0,
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 20,
                "flex": 1,
                "marginVertical": 20,
                "padding": 20,
              },
              {
                "alignItems": "center",
                "flex": undefined,
                "gap": 20,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Outfit",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  {
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              Curious about an animal?
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#919191",
                    "fontFamily": "Outfit",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              We will try to give you information based on what we have available...
            </Text>
          </View>
          <TextInput
            cursorColor="#FFFFFF"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="dot dot dot"
            placeholderTextColor="#919191"
            style={
              [
                {
                  "borderRadius": 9999,
                  "color": "#FFFFFF",
                  "fontFamily": "Outfit",
                  "fontSize": 18,
                  "paddingHorizontal": 20,
                  "paddingVertical": 8,
                  "width": "100%",
                },
                {
                  "elevation": 5,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                },
                {
                  "backgroundColor": "#393939",
                },
                undefined,
              ]
            }
            testID="search_input"
            value=""
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#327449",
                "borderRadius": 9999,
                "elevation": 5,
                "flexDirection": "row",
                "gap": 10,
                "justifyContent": "center",
                "margin": "auto",
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
            testID="search_btn"
          >
            <Text />
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "Outfit",
                    },
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    undefined,
                    {
                      "fontSize": 20,
                    },
                    undefined,
                  ]
                }
              >
                Search
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
